src/docfx/build/page/BuildPage.cs

```diff
@@ -50,7 +50,7 @@ public static async Task<List<Error>> Build(Context context, Document file)
                 ExtensionData = metadata,
             };
 
-            if (context.PublishModelBuilder.TryAdd(file, publishItem))
+            if (context.PublishModelBuilder.TryAdd(file, publishItem) && !context.Config.DryRun)
```

Comment:
is it possible to have a dry-run pipeline and a complete one instead of all the if check?

Reply:
Having two pipelines is harder to manage. I am trying to split actions that affect error log from those that don't.